modules/certificate/certificate.service.js

```diff
@@ -0,0 +1,67 @@
+const Certificate = require('./certificate.model');
+const RuleError = require('../../errors/rule.error');
+
+const {
+  STATUS_CODES: { NOT_FOUND },
+} = require('../../consts/status-codes');
+const {
+  CERTIFICATE_NOT_FOUND,
+} = require('../../error-messages/certificate.messages');
+
+class CertificatesService {
+  async getAllCertificates(skip, limit, user) {
+    let filter;
+    if (user?.role === 'admin' || user?.role === 'superadmin') {
+      filter = {};
+    } else {
+      filter = { createdBy: user.id };
+    }
+    const items = await Certificate.find(filter)
+      .sort({ startDate: 1 })
+      .limit(limit)
+      .skip(skip)
+      .exec();
+
+    const count = Certificate.find(filter).countDocuments();
+
+    return {
+      items,
+      count,
+    };
+  }
+
+  async getCertificateById(definedArg) {
+    const certificate = await Certificate.findById(definedArg).exec();
+    if (!certificate) {
+      throw new RuleError(CERTIFICATE_NOT_FOUND, NOT_FOUND);
+    }
+    return certificate;
+  }
+
+  async addCertificate(name, value, id) {
```

Comment:
done with findOne, but think it would better to use Mongoose's unique option in the certificate model and catch error 11011. https://masteringjs.io/tutorials/mongoose/unique